javalanglib/src/main/scala/java/lang/Class.scala

```diff
@@ -32,9 +33,22 @@ private trait ScalaJSClassData[A] extends js.Object {
 }
 
 final class Class[A] private (data0: Object) extends Object {
-  private val data: ScalaJSClassData[A] =
+  private[this] val data: ScalaJSClassData[A] =
     data0.asInstanceOf[ScalaJSClassData[A]]
 
+  /** Acces to `data` for other instances or `@inline` methods.
+   *
+   *  Directly accessing the `data` field from `@inline` methods will cause
+   *  scalac to make the field public and mangle its name. Since the Emitter
+   *  relies on the field being called exactly `data` in some of its
+   *  optimizations, we must avoid that.
+   *
+   *  This non-`@noinline` method can be used to access the field without
+   *  triggering scalac's mangling. Since it is a trivial accessor, the
+   *  Scala.js optimizer will inline it anyway.
+   */
+  private def getData(): ScalaJSClassData[A] = data
```

Comment:
I had to introduce this method. Since its comment why it was necessary.

Reply:
Hah, I would not have found this so quickly :P I also don't really understand where this is the case (that the `Emitter` needs to know this name). Could you point me to the code?

Reply:
Oh, I found it. The `zero` and `classDataOf` helpers fall back to this. This seems to be an inherent property of the optimization (that we want to retrieve the data again). So it seems we are stuck with it.

(An alternative would be to expose a public method, but because we are too late in the flow, we wouldn't get inlining, etc.)

Reply:
Yes, that's exactly the code and the problem.